common/beerocks/bwl/unit_tests/nl80211_client_test.cpp

```diff
@@ -0,0 +1,92 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "bwl/nl80211_client.h"
+
+#include <gtest/gtest.h>
+
+TEST(nl80211_client_test, get_channel_utilization_should_succeed_with_100_percent)
+{
+    bwl::nl80211_client::sChannelSurveyInfo channel_survey_info;
+    channel_survey_info.in_use       = true;
+    channel_survey_info.time_on_ms   = 100;
+    channel_survey_info.time_busy_ms = 100;
+
+    bwl::nl80211_client::SurveyInfo survey_info{{channel_survey_info}};
+
+    uint8_t channel_utilization;
+    ASSERT_TRUE(survey_info.get_channel_utilization(channel_utilization));
+    ASSERT_EQ(UINT8_MAX, channel_utilization);
+}
+
+TEST(nl80211_client_test, get_channel_utilization_should_succeed_with_50_percent)
+{
+    bwl::nl80211_client::sChannelSurveyInfo channel_survey_info;
+    channel_survey_info.in_use       = true;
+    channel_survey_info.time_on_ms   = 100;
+    channel_survey_info.time_busy_ms = 50;
+
+    bwl::nl80211_client::SurveyInfo survey_info{{channel_survey_info}};
+
+    uint8_t channel_utilization;
+    ASSERT_TRUE(survey_info.get_channel_utilization(channel_utilization));
+    ASSERT_EQ(UINT8_MAX / 2, channel_utilization);
+}
+
+TEST(nl80211_client_test, get_channel_utilization_should_succeed_with_0_percent)
+{
+    bwl::nl80211_client::sChannelSurveyInfo channel_survey_info;
+    channel_survey_info.in_use       = true;
+    channel_survey_info.time_on_ms   = 100;
+    channel_survey_info.time_busy_ms = 0;
+
+    bwl::nl80211_client::SurveyInfo survey_info{{channel_survey_info}};
+
+    uint8_t channel_utilization;
+    ASSERT_TRUE(survey_info.get_channel_utilization(channel_utilization));
+    ASSERT_EQ(0, channel_utilization);
+}
```

Comment:
I would normally write a helper function here to make the test instances simpler. I'm not sure if this is possible with googletest, but I would think so...
```suggestion
static void nl80211_client_test_get_channel_utilization_should_succeed(uint64_t time_busy_ms, uint8_t expected_channel_utilization)
{
    bwl::nl80211_client::sChannelSurveyInfo channel_survey_info;
    channel_survey_info.in_use       = true;
    channel_survey_info.time_on_ms   = 100;
    channel_survey_info.time_busy_ms = time_busy_ms;

    bwl::nl80211_client::SurveyInfo survey_info{{channel_survey_info}};

    uint8_t channel_utilization;
    ASSERT_TRUE(survey_info.get_channel_utilization(channel_utilization));
    ASSERT_EQ(expected_channel_utilization, channel_utilization);
}

TEST(nl80211_client_test, get_channel_utilization_should_succeed_with_0_percent)
{
    nl80211_client_test_get_channel_utilization_should_succeed(0, 0);
}
```

Reply:
It's possible, but there are some pitfalls to be aware of when using assertions in sub-routines:
- [Using Assertions in Sub-routines](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#using-assertions-in-sub-routines): 
"if a test sub-routine is called from several places, when an assertion inside it fails, it can be hard to tell which invocation of the sub-routine the failure is from. You can alleviate this problem using extra logging or custom failure messages, but that usually clutters up your tests." Google suggests the usage of the SCOPED_TRACE macro to qualify failure messages.
- [Asserting on Subroutines](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#asserting-on-subroutines): "if your test calls a subroutine that has an ASSERT_* failure in it, the test will continue after the subroutine returns. This may not be what you want.". 

Neither of those is, however, an issue in your suggested change. 

I tend to prefer simpler, self-contained and more readable tests at the expense of duplicating a few lines of code. But this is just an opinion. We should probably define and agree on a set of rules to write unit tests (to be discussed together with @RanRegev's [other test-related suggested topics](https://github.com/prplfoundation/prplMesh/pull/1321#pullrequestreview-415286769))